src/Lykke.Job.SiriusDepositsDetector/Controllers/MaintenanceController.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Common.Log;
+using Lykke.Common.Log;
+using Lykke.Job.SiriusDepositsDetector.ApiModels;
+using Lykke.Job.SiriusDepositsDetector.Services;
+using Lykke.Job.SiriusDepositsDetector.Settings;
+using Lykke.Service.Assets.Client;
+using Microsoft.AspNetCore.Mvc;
+using Swisschain.Sirius.Api.ApiClient;
+using Swisschain.Sirius.Api.ApiContract.Deposit;
+
+namespace Lykke.Job.SiriusDepositsDetector.Controllers
+{
+    [Route("api/maintenance")]
+    public class MaintenanceController : ControllerBase
+    {
+        private readonly IApiClient _apiClient;
+        private readonly IAssetsServiceWithCache _assetsService;
+        private readonly DepositProcessor _depositProcessor;
+        private readonly long _brokerAccountId;
+        private ILog _log;
+
+        public MaintenanceController(
+            ILogFactory logFactory,
+            IApiClient apiClient,
+            IAssetsServiceWithCache assetsService,
+            SiriusApiServiceClientSettings siriusClientSettings,
+            DepositProcessor depositProcessor)
+        {
+            _apiClient = apiClient;
+            _assetsService = assetsService;
+            _depositProcessor = depositProcessor;
+            _brokerAccountId = siriusClientSettings.BrokerAccountId;
+            _log = logFactory.CreateLog(this);
+        }
+
+        [HttpPost("process-deposit")]
+        public async Task<ActionResult> ProcessDeposit([FromBody] ProcessDepositRequest request, CancellationToken cancellationToken)
+        {
+            _log.Info("Manual processing of the deposit is being started", context: new
+            {
+                SiriusDepositId = request.SiriusDepositId
+            });
+
+            var assets = await _assetsService.GetAllAssetsAsync(false, cancellationToken);
+
+            var siriusRequest = new DepositUpdateSearchRequest
+            {
+                StreamId = Guid.NewGuid().ToString(),
+                BrokerAccountId = _brokerAccountId,
+                DepositId = request.SiriusDepositId,
+                State = 
+                {
+                    DepositState.Confirmed
+                }
+            };
+                           
+            _log.Info("Getting updates...", context: new
+            {
+                StreamId = siriusRequest.StreamId,
+                Cursor = siriusRequest.Cursor
+            });
+
+            var stream = _apiClient.Deposits.GetUpdates(siriusRequest);
+
+            if(await stream.ResponseStream.MoveNext(cancellationToken).ConfigureAwait(false))
```

Comment:
Shouldn't it be inverted? If !MoveNext (i.e. no elements) then nothing found? Or the semantics is different here?